lib/rubygems/gemcutter_utilities.rb

```diff
@@ -296,6 +301,18 @@ def webauthn_verification_url(credentials)
     response.is_a?(Net::HTTPSuccess) ? response.body : nil
   end
 
+  def default_browser_safari?
+    return false unless RUBY_PLATFORM.include?("darwin")
+
+    require "json"
+
+    launch_services = JSON.parse(`plutil -convert json -o - \
```

Comment:
`defaults read com.apple.LaunchServices/com.apple.launchservices.secure LSHandlers | plutil -convert json -o - -` is going to be safer than hardcoding paths to preferences